docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -12,14 +12,157 @@ xref:building-native-image.adoc[Building a Native Executable],
+increase the reliability and improve the runtime performance of native executables.
+These are the high level sections to be found in this guide:
+
+* <<native-memory-management,Native Memory Management>>
+* <<inspecting-and-debugging,Inspecting and Debugging Native Executables>>
+* <<native-faq,Frequently Asked Questions>>
+
+[[native-memory-management]]
+== Native Memory Management
+Memory management for Quarkus native executables is enabled by GraalVM’s SubstrateVM runtime system.
+The memory management component in GraalVM is explained in detail
+link:https://www.graalvm.org/22.2/reference-manual/native-image/optimizations-and-performance/MemoryManagement[here].
+This guide complements the information in the GraalVM website with further observations particularly relevant for Quarkus applications.
+
+=== Garbage Collectors
+The garbage collectors available for Quarkus users are currently Serial GC and Epsilon GC.
+
+==== Serial GC
+Serial GC, the default option in GraalVM, is a single-threaded non-concurrent GC, just like HotSpot’s Serial GC.
+The implementation in GraalVM however is different to the HotSpot one,
+and there can be significant differences in the runtime behavior.
+
+One of the key differences between HotSpot’s Serial GC and GraalVM’s Serial GC is precisely on how full GCs are executed.
+In HotSpot the algorithm used is mark-sweep-compact whereas in GraalVM it is mark-copy.
+Both need to traverse all live objects,
+but in mark-copy this traversal is also used to copy live objects to a secondary space or semi-space.
+As objects are copied from one semi-space to another they’re also compacted.
+In mark-sweep-compact, the compacting requires a second pass on the live objects.
+This makes full GCs in mark-copy more time efficient (in terms of time spent in each GC cycle) than mark-sweep-compact.
+The tradeoff mark-copy makes in order to make individual full GC cycles shorter is space.
+The use of semi-spaces means that for an application to maintain the same GC performance that mark-sweep achieves (in terms of allocated MB per second),
+it requires double the amount of memory.
+
+Just like in GraalVM, Quarkus also uses Serial GC as the default gargabe collector.
+
+===== GC Collection Policy
+The default GC collection policy is called “adaptive”,
+and it is based on HotSpot's ParallelGC adaptive size policy.
+The main difference with HotSpot is its focus on memory footprint.
+This means that GraalVM’s adaptive GC policy tries to aggressively trigger GCs in order to keep memory consumption down.
+
+Up to version 2.13, Quarkus used “space/time” GC collection policy,
+but starting with version 2.14, Quarkus uses “adaptive” policy.
+Quarkus chose to use space/time early on because it had considerable performance improvements over adaptive,
+but recently we have discovered that space/time can have a worse out-of-the-box experience compared with space/time,
+while at the same time the benefits it offered have diminished considerably.
```

Comment:
```suggestion
The reason why Quarkus initially chose to use "space/time" is because at that time it had considerable performance improvements over "adaptive".
Recent performance experiments, however, indicate that the "space/time" policy can result in worse out-of-the-box experience compared to the "space/time" policy,
while at the same time the benefits it used to offer have diminished considerably after improvements made to the "adaptive" policy.
```